app/models/activity_logger.rb

```diff
@@ -26,15 +33,18 @@ class ActivityLogger
   # 3) when the activity is complete,
   #    -- the log file will be closed automatically if opened with a block, OR
   #    -- call log.close
+  #
+  # NOTE: this requires ActiveSupport.  It can only be run with Rails
 
   def self.open(filename, facility, activity, show=true)
+
     log = new(filename, facility, activity, show)
 
     if block_given?
       begin
         yield log
       ensure
-        log.close
+        log.close unless @is_system_outstream
```

Comment:
I don't think we need to put the boolean qualifier on this - also, I don't think we want to, either, since we'd still like to see the closing messages (time finished, logged event duration) even if writing to `$stdout`.

Reply:
I tried that first, but it caused *huge* problems.  Anything else that tried to write to stdout reported "... closed stream".

(You're probably thinking the same thing that I did: that anything using stdout would try to open it if it was closed.  but that's not how it works, apparently.)

Reply:
@patmbolger I don't think it will work.  Give it a try and let me know.  (It's possible that I'm doing something wrong or overlooking something.)

Reply:
@weedySeaDragon yes, I see the same problem.  

If the user does not use a block, there are two issues:
1.  The output appears twice - once for the "file" (stdout) and, of course, stdout (via puts):
```
 log = ActivityLogger.open('log/testing3.log', 'test', 'test')
pry(main)> log.record('info', 'this is a test message')  
[test] [test] [info] this is a test message
[test] [test] [info] this is a test message
```

2) Same problem when `close` is called explicitly by the user:
```
log.close
[test] [test] [info] Finished at 2018-11-30 23:04:56 UTC.
[test] [test] [info] Finished at 2018-11-30 23:04:56 UTC.
[test] [test] [info] Duration: 44.37 seconds.

[test] [test] [info] Duration: 44.37 seconds.
IOError: closed stream
from /Users/pbolger/Desktop/shf-project/app/models/activity_logger.rb:91:in `write'
Traceback (most recent call last):
	34: from bin/rails:4:in `<main>'
	33: from /Users/pbolger/.rvm/gems/ruby-2.5.1/gems/activesupport-5.2.1/lib/active_support/dependencies.rb:287:in `require'
	32: from /Users/pbolger/.rvm/gems/ruby-2.5.1/gems/activesupport-5.2.1/lib/active_support/dependencies.rb:253:in `load_dependency'
	31: from /Users/pbolger/.rvm/gems/ruby-2.5.1/gems/activesupport-5.2.1/lib/active_support/dependencies.rb:287:in `block in require'
	30: from /Users/pbolger/.rvm/gems/ruby-2.5.1/gems/bootsnap-1.3.2/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:29:in `require'
	29: from /Users/pbolger/.rvm/gems/ruby-2.5.1/gems/bootsnap-1.3.2/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:20:in `require_with_bootsnap_lfi'
	28: from /Users/pbolger/.rvm/gems/ruby-2.5.1/gems/bootsnap-1.3.2/lib/bootsnap/load_path_cache/loaded_features_index.rb:65:in `register'
	27: from /Users/pbolger/.rvm/gems/ruby-2.5.1/gems/bootsnap-1.3.2/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:21:in `block in require_with_bootsnap_lfi'
	26: from /Users/pbolger/.rvm/gems/ruby-2.5.1/gems/bootsnap-1.3.2/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:21:in `require'

.... etc ....
```